docs/development.rst

```diff
@@ -24,15 +24,28 @@ Publishing
 
 Publishing to `PyPI <https://pypi.org/project/pyhf/>`__ and `TestPyPI <https://test.pypi.org/project/pyhf/>`__
 is automated through the `PyPA's PyPI publish GitHub Action <https://github.com/pypa/gh-action-pypi-publish>`__.
-To publish a release to PyPI one simply needs to run
+Publishing a release to PyPI begins with creating a new branch :code:`release/cut-<release tag>`
+and then from that branch running
 
 .. code-block:: console
 
     bumpversion [major|minor|patch]
 
-to update the release version and get a tagged commit and then push the commit
-and tag to :code:`master` with
+to update the release version and get a tagged commit.
+Then, push only the branch to GitHub and open a PR to ensure that the code
+you are going to publish is indeed stable.
+Once the PR has passed all checks and has been approved, a maintainer should merge
+the branch from the command line
 
 .. code-block:: console
 
-    git push origin master --tags
+    git checkout master
+    git merge release/cut-<release tag>
+
+and then push the commit and tag to :code:`master` with
+
+.. code-block:: console
+
+    git push origin master <release tag>
```

Comment:
The idea is that you push the tag first, before merging the PR and if the publishing happens properly, only then do the merge.

Reply:
Ah, sorry. Yes, this is a **_very_** good point, and apparently not clear to me late at night while working on this. Thank you for pointing this out!